utils/log.py

```diff
@@ -476,7 +476,7 @@ def art_log(self, level_name, message, kwargs):
             message = str(message)
         art_log_record = {
             'level': level_name,
-            'message': message.encode("ascii", "xmlcharrefreplace"),
+            'message': message.decode('utf-8').encode("ascii", "xmlcharrefreplace"),
```

Comment:
For this, I think the correct approach would be:
* If the message is not a basestring, inspect it if it has `__unicode__`. If yes, do  `unicode()` on it and continue with step 2. If it does not have `__unicode__`, do `str()` on it and continue with step 3.
* check if message is unicode, if yes, encode with xmlcharrefreplace
* if message is a str, do `.decode("utf-8", "ignore").encode("ascii", "xmlcharrefreplace")`

This should ensure proper handling of any string, unicode or any object passed to the function.

@psav ?